tests/model_loading/test_locate.py

```diff
@@ -58,6 +58,16 @@ def test_found_when_passed_path(repo_mock):
     assert loader.path == PosixPath(dir_path)
 
 
+def test_found_when_passed_bucket_root_as_path(repo_mock):
+    bucket_name = "some-bucket/random/prefix"
+    repo_mock.add_storage("s3", bucket_name)
+    repo_mock.add_storage_file(f"s3/{bucket_name}/model.pt", b"blablabla")
+    locator = ModelLocator(repo_mock, path=bucket_name)
```

Comment:
To see the bug, you can try this additional test case
`path=bucket_name + "bla"`